### ISSUE TITLE
Question about "cocos2d: removeAction: Target not found"

### ISSUE THREAD

**Issue by doyoubi** [cocos2d/cocos2d-x#9019]
I'm using cocos2dx v3.2.  
When there is already an Action running on a sprite and we call sprite->runAction(another Action), the program will crash.  
So I add sprite->stopAllActions() before run another Animate. But this will cause the error message "cocos2d: removeAction: Target not found" arise again and again.  

The code of ActionManager::removeAction is:  

```
void ActionManager::removeAction(Action *action)
{
    // explicit null handling
    if (action == nullptr)
    {
        return;
    }

    tHashElement *element = nullptr;
    Ref *target = action->getOriginalTarget();
    HASH_FIND_PTR(_targets, &target, element);
    if (element)
    {
        auto i = ccArrayGetIndexOfObject(element->actions, action);
        if (i != CC_INVALID_INDEX)
        {
            removeActionAtIndex(i, element);
        }
    }
    else
    {
        CCLOG("cocos2d: removeAction: Target not found");
    }
}
```

When the error message occur, removeAction is invoked in ActionManager::update  

```
......
if (_currentTarget->currentActionSalvaged)
{
    // The currentAction told the node to remove it. To prevent the action from
    // accidentally deallocating itself before finishing its step, we retained
    // it. Now that step is done, it's safe to release it.
    _currentTarget->currentAction->release();
} else
if (_currentTarget->currentAction->isDone())
{
    _currentTarget->currentAction->stop();

    Action *action = _currentTarget->currentAction;
    // Make currentAction nil to prevent removeAction from salvaging it.
    _currentTarget->currentAction = nullptr;
    removeAction(action);              // invoked here
}
......
```

I don't know whether it is a bug.
In the Chinese offical community, there is a thread about this problem, http://www.cocoachina.com/bbs/read.php?tid=176181  
The offical said that it will not cause crash so we can just ignore it.  
I can't understand this code completely, but it seems that after I call sprite->stopAllActions(), the engine can't clear the previous running Action correctly, the `_currentTarget->currentAction` still own the pointer of previous running Action.
Is it a bug? How can I solve this problem?

**Comment by pkxpp** [comment]
I have this problem, too.

**Comment by jianglong0156** [comment]
the bug had already fix in 3.8, Please test again
